plugins/default-browser-emulator/index.ts

```diff
@@ -51,7 +52,19 @@ export default class DefaultBrowserEmulator extends BrowserEmulatorBase {
 
   constructor(createOptions: IPluginCreateOptions, userAgentOption: IUserAgentOption) {
     super(createOptions, userAgentOption);
-    this.data = dataLoader.as(userAgentOption);
+    this.data = dataLoader.as(userAgentOption) as any;
+
+    this.data.deviceMemory = Math.ceil(Math.random() * 4) * 2;
+    this.data.videoDevice = {
+      deviceId: randomBytes(32).toString('hex'),
+      groupId: randomBytes(32).toString('hex'),
+    };
+    this.data.webGLParameters = {
+      // UNMASKED_VENDOR_WEBGL
+      37445: 'Intel Inc.',
+      // UNMASKED_RENDERER_WEBGL
+      37446: 'Intel Iris OpenGL Engine',
+    };
```

Comment:
This new code seems to fundamentally changes the clarity of `data` from being the "imported emulator data files for a browser" to something that we can hang extra data on if we don't know a better place to put it.

I'm not against the change, and I definitely don't have a better alternative at the moment, so I'm inclined to go with it. Just putting it out there in case it spurs some fresh ideas.

Reply:
I kind of recognized that you were doing that and kind of didnt’. Maybe if we called it ‘generatedData’ or something, that would help at least differentiate?
2 options I can think of:
1. We can make the emulator builder back into a single instance per plugin instance. You changed it to be created everytime a new page or new worker is created. I wasn’t entirely sure why, but I remember you mentioning it, so I left it
2. We could have the idea of a “userProfile” or “computerProfile” include things like this - device id, gpu, memory, etc. we could have it be the second parameter into those helpers